### PR TITLE
[OMP][stable-3.3.0] #6610 Add support for displaying export results for ONIX plugin

### PR DESCRIPTION
**Pull request by defstat**

### CODE REVIEW

File: plugins/importexport/onix30/Onix30ExportPlugin.inc.php [hunk 1]
```diff
@@ -101,18 +102,50 @@ function display($args, $request) {
...
+
+				$templateMgr->assign('errorsAndWarnings', $problems);
+				$templateMgr->assign('errorsFound', $foundErrors);
+				$templateMgr->assign('onixPlugin', $this);
+
+				$json = new JSONMessage(true, $templateMgr->fetch($this->getTemplateResource('resultsExport.tpl')));
+				header('Content-Type: application/json');
+				return $json->getString();
+			case 'downloadExportFile':
+				$downloadPath = $request->getUserVar('downloadFilePath');
```
Comment (asmecher) [inline plugins/importexport/onix30/Onix30ExportPlugin.inc.php:142]: This is an arbitrary "download a file then delete it" feature, I think :scream:

File: plugins/importexport/onix30/templates/resultsExport.tpl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	<div id="export-tab">
+		<script type="text/javascript">
+			$(function() {ldelim}
+				// Attach the form handler.
+				$('#exportXmlForm').pkpHandler('$.pkp.controllers.form.FormHandler');
+			{rdelim});
+		</script>
+		<form id="exportXmlForm" class="pkp_form" action="{plugin_url path="downloadExportFile"}" method="post">
+			{csrf}
+			<input type="hidden" name="downloadFilePath" id="downloadFilePath" value="{$exportPath}" />
```
Comment (asmecher) [inline plugins/importexport/onix30/templates/resultsExport.tpl:25]: `$exportPath` should be passed through `|escape` -- but my other comment about passing in `downloadFilePath` is more important to fix.

File: plugins/importexport/onix30/Onix30ExportPlugin.inc.php (hunk 1 above)
Comment (defstat) [reply plugins/importexport/onix30/Onix30ExportPlugin.inc.php:143]: You are right @asmecher.. The user can "play" with the hidden field to download what ever (and delete it afterwards).

I am proposing a fix for that, though not sure if its the correct approach: The code is now checking if the file that needs to be downloaded is in the "correct" export path, and also if the name matches the name that the plugin expects to be.

File: plugins/importexport/onix30/templates/resultsExport.tpl (hunk 2 above)
Comment (defstat) [reply plugins/importexport/onix30/templates/resultsExport.tpl]: Changed.

File: plugins/importexport/onix30/Onix30ExportPlugin.inc.php (hunk 1 above)
Comment (asmecher) [reply plugins/importexport/onix30/Onix30ExportPlugin.inc.php:143]: @defstat, if the handling code already knows the correct export filename, then why do we need to bounce it off the user in the first place? Are there several possible valid filenames?

Comment (defstat) [reply plugins/importexport/onix30/Onix30ExportPlugin.inc.php:147]: Thanks @asmecher ... The code now passes only the date part of the file in order to determine the file that will be downloaded

File: plugins/importexport/onix30/Onix30ExportPlugin.inc.php [hunk 3]
```diff
@@ -101,18 +104,67 @@ function display($args, $request) {
...
+				$templateMgr->assign('onixPlugin', $this);
+
+				$json = new JSONMessage(true, $templateMgr->fetch($this->getTemplateResource('resultsExport.tpl')));
+				header('Content-Type: application/json');
+				return $json->getString();
+			case 'downloadExportFile':
+				if (!$request->checkCSRF()) throw new Exception('CSRF mismatch!');
+				$downloadPath = $request->getUserVar('downloadFilePath');
+				
+				$date = DateTime::createFromFormat($exportFileDatePartFormat, $downloadPath);
```
Comment (asmecher) [inline plugins/importexport/onix30/Onix30ExportPlugin.inc.php:149]: I think you could skip the double-validating (both here and in `getExportFilename`) if the new parameter to `getExportFilename` was a `?DateTime`  object. Then you could validate here when calling `DateTime::createFromFormat`, and within `getExportFilename`, just format to a string when the parameter is provided.

**Pull request merged**